pygeos/_geometry.pyx

```diff
@@ -75,3 +79,114 @@ def get_parts(object[:] array):
                 idx += 1
 
     return parts, index
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def collections_1d(
+    object geometries,
+    object indices,
+    int geometry_type = 7,
+    int ndim = 2
+):
+    cdef Py_ssize_t geom_idx_1 = 0
+    cdef Py_ssize_t coll_idx = 0
+    cdef Py_ssize_t coll_size = 0
+    cdef Py_ssize_t coll_geom_idx = 0
+    cdef GEOSGeometry *geom = NULL
+    cdef GEOSGeometry *coll = NULL
+    cdef int expected_type = -1
+    cdef int expected_type_alt = -1
+    cdef int curr_type = -1
+
+    if geometry_type == 4:  # MULTIPOINT
+        expected_type = 0
+    elif geometry_type == 5:  # MULTILINESTRING
+        expected_type = 1
+        expected_type_alt = 2
+    elif geometry_type == 6:  # MULTIPOLYGON
+        expected_type = 3
+    elif geometry_type == 7:
+        pass
+    else:
+        raise ValueError(f"Invalid geometry_type: {geometry_type}.")
+
+    # Cast input arrays and define memoryviews for later usage
+    geometries = np.asarray(geometries, dtype=np.object)
+    if geometries.ndim != 1:
+        raise TypeError("geometries is not a one-dimensional array.")
+
+    indices = np.asarray(indices, dtype=np.int32)
+    if indices.ndim != 1:
+        raise TypeError("indices is not a one-dimensional array.")
+
+    if geometries.shape[0] != indices.shape[0]:
+        raise ValueError("geometries and indices do not have equal size.")
+
+    if geometries.shape[0] == 0:
+        # return immediately if there are no geometries to return
+        return np.empty(shape=(0, ), dtype=np.object_)
+
+    if np.any(indices[1:] < indices[:-1]):
```

Comment:
This is raising a Cython warning
> warning: pygeos/_geometry.pyx:130:38: the result of using negative indices inside of code sections marked as 'wraparound=False' is undefined

Probably need to avoid using `@cython.wraparound(False)`.